app/gui/view/component-browser/breadcrumbs/src/entry.rs

```diff
@@ -0,0 +1,343 @@
+            let background = Rect((background_width.px(), background_height.px()));
+            let background = background.corners_radius(background_corners_radius.px());
+            let background = background.fill(background_color);
+            let shape = background + circles;
+            shape.into()
+        }
+    }
+}
+
+
+
+// =============
+// === Model ===
+// =============
+
+/// A model for the entry in the breadcrumbs list.
+#[allow(missing_docs)]
+#[derive(Clone, CloneRef, Debug, Default)]
+pub enum Model {
+    Text(ImString),
+    Separator,
+    #[default]
+    Ellipsis,
+}
+
+#[allow(missing_docs)]
+#[derive(Clone, Copy, CloneRef, Debug, Default, PartialEq)]
+enum State {
+    Text,
+    Separator,
+    #[default]
+    Ellipsis,
+}
+
+
+
+// =============
+// === Entry ===
+// =============
+
+// === EntryData ===
+
+/// An internal structure of [`Entry`], which may be passed to FRP network.
+#[allow(missing_docs)]
+#[derive(Clone, Debug)]
+pub struct EntryData {
+    display_object: display::object::Instance,
+    text:           text::Area,
+    separator:      separator::View,
+    ellipsis:       ellipsis::View,
+    state:          Rc<Cell<State>>,
+}
+
+impl EntryData {
+    fn new(app: &Application, text_layer: Option<&Layer>) -> Self {
+        let logger = Logger::new("breadcrumbs::Entry");
+        let display_object = display::object::Instance::new(&logger);
+        let text = app.new_view::<ensogl_text::Area>();
+        if let Some(layer) = text_layer {
+            text.add_to_scene_layer(layer);
+        }
+        let ellipsis = ellipsis::View::new(&logger);
+        let separator = separator::View::new(&logger);
+        let state = default();
+        display_object.add_child(&ellipsis);
+        Self { display_object, state, text, ellipsis, separator }
+    }
+
+    fn remove_current(&self) {
+        match self.state.get() {
+            State::Text => self.text.unset_parent(),
+            State::Separator => self.separator.unset_parent(),
+            State::Ellipsis => self.ellipsis.unset_parent(),
+        }
+    }
+
+    fn set_model(&self, model: &Model) {
+        match model {
+            Model::Text(content) => self.set_content(content),
+            Model::Separator => self.set_separator(),
+            Model::Ellipsis => self.set_ellipsis(),
+        }
+    }
+
+    fn set_content(&self, content: &str) {
+        self.text.set_content(content);
+        if self.state.get() != State::Text {
+            self.remove_current();
+            self.display_object.add_child(&self.text);
+            self.state.set(State::Text);
+        }
+    }
+
+    fn set_separator(&self) {
+        if self.state.get() != State::Separator {
+            self.remove_current();
+            self.display_object.add_child(&self.separator);
+            self.state.set(State::Separator);
+        }
+    }
+
+    fn set_ellipsis(&self) {
+        if self.state.get() != State::Ellipsis {
```

Comment:
these names are not explanatory. What does `Entry::set_separator` mean? It sounds like I will be setting a separator in an entry. I mean, I know wht it means, but I feel that a better name can be found here (I dont know yet what name, but I'd be thankful if you think about it).

Reply:
Renamed to `switch_to_...`.